core/templates/dev/head/components/profileLink.js

```diff
@@ -0,0 +1,21 @@
+oppia.directive('profileLinkText', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      linkedName: '@'
+    },
+    templateUrl: 'components/profileLinkText',
+  };
+}]);
+
+oppia.directive('profileLinkImage', [function() { 
+  return {
+    restrict: 'E',
+    transclude: true,
+    scope: {
+      linkedName: '@',
+      linkedImage: '@'
```

Comment:
I wouldn't transclude. Couldn't you build the image url based on the linkedName that's passed in? Grepping the backend, I see a ProfilePictureHandler, and looking in main.py it looks like it's accessible via /preferenceshandler/profile_picture. You could make a $http call for this.